src/toolchain.rs

```diff
@@ -44,11 +45,23 @@ pub enum UpdateStatus {
 
 impl<'a> Toolchain<'a> {
     pub fn from(cfg: &'a Cfg, name: &str) -> Result<Self> {
+        let v: Vec<&str> = name.split('/').collect();
+        let name = v[0];
+
         let resolved_name = cfg.resolve_toolchain(name)?;
         let path = cfg.toolchains_dir.join(&resolved_name);
+
+        let resolved_name_dated = if v.len() > 1 {
+            let name_dated = name.to_string() + "-" + v[1];
```

Comment:
This approach to adding the date will fail if any of the platform details are included in the part before the `/` for example, if I wanted nightly from date `YYYY-MM-DD` for platform `i686-pc-windows-gnu` then I'd plausibly type `rustup toolchain install nightly-i686-pc-windows-gnu/YYYY-MM-DD` which will result in an incorrectly formed toolchain descriptor.